src/Razor/src/Microsoft.NET.Sdk.Razor/GenerateAspNetCoreStaticAssetsManifest.cs

```diff
@@ -0,0 +1,179 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Xml;
+using System.Xml.Linq;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace Microsoft.AspNetCore.Razor.Tasks
+{
+    public class GenerateAspNetCoreStaticAssetsManifest : Task
+    {
+        private const string ContentRoot = nameof(ContentRoot);
+        private const string BasePath = nameof(BasePath);
+        
+        [Required]
+        public string TargetManifestPath { get; set; }
+
+        [Required]
+        public ITaskItem[] ContentRootDefinitions { get; set; }
+
+        public override bool Execute()
+        {
+            if(!ValidateArguments())
+            {
+                return false;
+            }
+
+            return ExecuteCore();
+        }
+
+        private bool ExecuteCore()
+        {
+            var document = new XDocument(new XDeclaration("1.0", "utf-8", "yes"));
```

Comment:
Curious, why an XML manifest vs. doing something like an assembly attribute that took a list of static assets?

Reply:
Cuz it might not end-up being embedded and might end up on disk, given that this is a development feature only

Reply:
And XML just because it's netstandard

Reply:
> Cuz it might not end-up being embedded and might end up on disk

Interesting. I imagine that's the scenario where the RCL project doesn't build? 

Just to make sure I understand everything fully, in that scenario would the SDK have copied the manifest to the output despite the build failure? Assuming so, would the middleware be smart enough ignore the outdated embedded manifest file and look next to it it on disk?

Definitely curious in this area 😄 

> given that this is a development feature only

oh interesting, there's no manifest in production scenarios?

Reply:
No, during publish we will put the files in the wwwroot folder of the published app

Reply:
https://gist.github.com/javiercn/050cefbecc5f623630e611088ba5b891 if you are interested